local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -33,13 +34,17 @@ import (
 // It looks for volumes in the directories specified in the discoveryMap
 type Discoverer struct {
 	*common.RuntimeConfig
-	Labels          map[string]string
+	Labels map[string]string
+	// ProcTable is a reference to running processes so that we can prevent PV from being created while
+	// it is being cleaned
+	ProcTable       deleter.ProcTable
 	nodeAffinityAnn string
+	blockDisabled   bool
 }
 
 // NewDiscoverer creates a Discoverer object that will scan through
 // the configured directories and create local PVs for any new directories found
-func NewDiscoverer(config *common.RuntimeConfig) (*Discoverer, error) {
+func NewDiscoverer(config *common.RuntimeConfig, procTable deleter.ProcTable, blockDisabled bool) (*Discoverer, error) {
```

Comment:
Can you add it to the common.RuntimeConfig instead? It'll be easier to remove in the future.

Reply:
@msau42 Fixed.